### PR TITLE
Exporting FHIR bundle

### PR DESCRIPTION
**Pull request by RohitKandimalla**
image

Navigation - Measure Composer / Measure Packager / Create Package and Export Button

### CODE REVIEW

File: src/main/java/mat/client/measurepackage/MeasurePackagePresenter.java [hunk 1]
```diff
@@ -168,70 +168,58 @@ public void onClick(ClickEvent event) {
...
+                MatContext.get().getPackageService().saveRiskVariables(currentDetail, new AsyncCallback<Void>() {
+                    @Override
+                    public void onFailure(final Throwable caught) {
+                        logger.log(Level.SEVERE, "Error in PackageService.saveRiskVariables. Error message: " + caught.getMessage(), caught);
+                        if (caught instanceof SaveRiskAdjustmentVariableException) {
                             getMeasurePackageOverview(MatContext.get().getCurrentMeasureId());
-                            view.getRiskAdjustmentVariableSuccessMessageDisplay().createAlert(MatContext.get().getMessageDelegate().getRiskAdjSavedMessage());
-                            showMeasurePackagerBusy(false);
+                            view.getRiskAdjustmentVariableErrorMessageDisplay().createAlert(caught.getLocalizedMessage());
+                        } else {
```
Comment (serhii-ilin) [inline src/main/java/mat/client/measurepackage/MeasurePackagePresenter.java:209]: I would suggest adding more logging to onFailure and onSuccess

File: src/main/java/mat/server/humanreadable/HumanReadableGenerator.java [hunk 2]
```diff
@@ -69,7 +71,13 @@
 			MatConstants.MEASURE_POPULATION, MatConstants.MEASURE_POPULATION_EXCLUSIONS, MatConstants.STRATUM ,
 			MatConstants.MEASURE_OBSERVATION_POPULATION};
 	
-	@Autowired CQLHumanReadableGenerator humanReadableGenerator; 
+	@Autowired CQLHumanReadableGenerator humanReadableGenerator;
```
Comment (serhii-ilin) [inline src/main/java/mat/server/humanreadable/HumanReadableGenerator.java:74]: the line is not aligned properly

File: src/main/java/mat/model/clause/MeasureExport.java [hunk 3]
```diff
@@ -170,20 +170,20 @@ public void setFhirXml(String fhirXml) {
 	}
 
 	@Column(name = "FHIR_LIBS_XML")
-	public String getFhirLibsXml() {
+	public String getFhirIncludedLibsXml() {
 		return fhirLibsXml;
 	}
 
-	public void setFhirLibsXml(String fhirLibsXml) {
+	public void setFhirIncludedLibsXml(String fhirLibsXml) {
```
Comment (serhii-ilin) [inline src/main/java/mat/model/clause/MeasureExport.java:177]: Can we correct field names accordinlgy to their setters/getters?

File: src/main/java/mat/server/ExportServlet.java [hunk 4]
```diff
@@ -357,8 +359,8 @@ private void exportCQLLibraryFile(HttpServletResponse resp, String id, String ty
         ExportResult export = getService().createOrGetCQLLibraryFile(id, measureExport);
 
         if (!export.getIncludedCQLExports().isEmpty()) {
-            ZipPackager zp = new ZipPackager();
-            zp.getCQLZipBarr(export, "cql");
+            ZipPackager zp = context.getBean(ZipPackagerFactory.class).getZipPackager();
```
Comment (serhii-ilin) [inline src/main/java/mat/server/ExportServlet.java:362]: I wonder if we can extract a generic common logic in a single method with MIME type as a parameter

File: src/main/java/mat/server/service/impl/ZipPackager.java [hunk 5]
```diff
@@ -179,31 +212,65 @@ public void createBulkExportZip(String emeasureName,Date exportDate, byte[] wkbk
...
 	 */
-	public void getZipBarr(String emeasureName, ZipOutputStream zip,
-						 String packageDate,String emeasureHTMLStr, String simpleXmlStr, String emeasureXMLStr, ExportResult cqlExportResult, ExportResult elmExportResult, ExportResult jsonExportResult, String currentRealeaseVersion, String parentPath) {
+	public void getZipBarr(String emeasureName, ZipOutputStream zip,String emeasureHTMLStr,
+                           String emeasureXMLStr, ExportResult cqlExportResult, ExportResult elmExportResult, ExportResult jsonExportResult,
+                           String currentRealeaseVersion, String parentPath, String measureId) {
+
+        Measure measure = measureDAO.getMeasureByMeasureId(measureId);
+        MeasureExport measureExport = measureExportDAO.findByMeasureId(measureId);
+        CQLLibrary cqlLibrary = cqlLibraryDAO.getLibraryByMeasureId(measureId);
```
Comment (serhii-ilin) [inline src/main/java/mat/server/service/impl/ZipPackager.java:221]: something is wrong with the formatting of the class

File: src/main/java/mat/server/service/impl/ZipPackagerFactory.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package mat.server.service.impl;
+
+        import org.springframework.beans.factory.annotation.Lookup;
```
Comment (serhii-ilin) [inline src/main/java/mat/server/service/impl/ZipPackagerFactory.java:3]: formatting

File: src/main/java/mat/client/measurepackage/MeasurePackagePresenter.java (hunk 1 above)
Comment (RohitKandimalla) [reply src/main/java/mat/client/measurepackage/MeasurePackagePresenter.java:202]: Added couple of Logging statements, I didn’t work on this feature, re-formatted some code.

File: src/main/java/mat/server/ExportServlet.java (hunk 4 above)
Comment (RohitKandimalla) [reply src/main/java/mat/server/ExportServlet.java]: Created a Generic method for 3 different formats.

**Pull request merged**